public/app/features/dashboard-scene/saving/SaveDashboardDrawer.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+
+import { SceneComponentProps, SceneObjectBase, SceneObjectState, SceneObjectRef } from '@grafana/scenes';
+import { Dashboard } from '@grafana/schema';
+import { Drawer, Tab, TabsBar } from '@grafana/ui';
+import { SaveDashboardDiff } from 'app/features/dashboard/components/SaveDashboard/SaveDashboardDiff';
+
+import { DashboardScene } from '../scene/DashboardScene';
+import { transformSceneToSaveModel } from '../serialization/transformSceneToSaveModel';
+import { jsonDiff } from '../settings/version-history/utils';
+
+import { SaveDashboardAsForm } from './SaveDashboardAsForm';
+import { SaveDashboardForm } from './SaveDashboardForm';
+import { DashboardChangeInfo } from './shared';
+
+interface SaveDashboardDrawerState extends SceneObjectState {
+  dashboardRef: SceneObjectRef<DashboardScene>;
+  showDiff?: boolean;
+  saveTimeRange?: boolean;
+  saveVariables?: boolean;
+  saveAsCopy?: boolean;
+}
+
+export class SaveDashboardDrawer extends SceneObjectBase<SaveDashboardDrawerState> {
+  public onClose = () => {
+    this.state.dashboardRef.resolve().setState({ overlay: undefined });
+  };
+
+  public onToggleSaveTimeRange = () => {
+    this.setState({ saveTimeRange: !this.state.saveTimeRange });
+  };
+
+  public onToggleSaveVariables = () => {
+    this.setState({ saveTimeRange: !this.state.saveTimeRange });
+  };
+
+  public getSaveDashboardChange(): DashboardChangeInfo {
```

Comment:
If we extract this into shared/utils, we could easily unit test it. Not talking about the scene resolution, only the calculation of changes. thoughts?

Reply:
done